### PR TITLE
feat(Makefile): Default sanity checks for all repos

### PR DESCRIPTION
**Pull request by owlot**
Refactored the Makefile to split up in separate includes so they can be reused for different repository types.

This should also solve the issue that sometimes files with the wrong formatting was added through Terraform provisioning, which broke the pipelines in other projects.

### CODE REVIEW

File: src/repositories_rust.tf [hunk 1]
```diff
@@ -44,12 +44,18 @@ locals {
...
+      ".make/python.mk" = {
+        content = file("templates/all/.make/python.mk")
+      },
+      ".make/rust.mk" = {
+        content = file("templates/all/.make/rust.mk")
+      },
+      ".make/toml.mk" = {
+        content = file("templates/all/.make/toml.mk")
       },
       ".taplo.toml" = {
```
Comment (ServiceDog) [inline src/repositories_rust.tf:60]: cspell and editorconfig .mk?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+## DO NOT EDIT!
+# This file was provisioned by Terraform
+# File origin: https://github.com/Arrow-air/tf-github/tree/main/src/templates/all/Makefile
+
+DOCKER_NAME := arrow-${name}
+
+help: .help-base .help-cspell .help-markdown .help-editorconfig
```
Comment (ServiceDog) [inline Makefile:7]: toml, python, docker helps and includes?

File: src/repositories_templates.tf [hunk 3]
```diff
@@ -108,7 +108,7 @@ module "repository_svc_template" {
   repository_files = merge(
     try(each.value.files, {}),
     { for file, path in local.template_files :
-      file => { content = templatefile(path, { owner_team = each.value.owner_team }) }
+      file => { content = templatefile(path, { owner_team = each.value.owner_team, name = format("lib-template-%s", each.key) }) }
```
Comment (ServiceDog) [inline src/repositories_templates.tf:111]: *svc-template

File: src/templates/all/Makefile [hunk 4]
```diff
@@ -0,0 +1,14 @@
+## DO NOT EDIT!
+# This file was provisioned by Terraform
+# File origin: https://github.com/Arrow-air/tf-github/tree/main/src/templates/all/Makefile
+
+DOCKER_NAME := arrow-${name}
+
+help: .help-base .help-cspell .help-markdown .help-editorconfig
```
Comment (ServiceDog) [inline src/templates/all/Makefile:7]: toml, python, rust, docker helps and includes?

File: src/templates/rust-all/Makefile.tftpl [hunk 5]
```diff
@@ -1,207 +1,36 @@
...
-	@docker kill $(DOCKER_NAME)-run || true
+export
 
+help: .help-base .help-rust .help-python .help-cspell .help-markdown .help-editorconfig .help-toml .help-docker
 all: test build release docker-build
+
+include .make/docker.mk
 %{ endif }
+
+test: rust-check rust-test rust-clippy rust-fmt toml-test python-test md-test-links
```
Comment (ServiceDog) [inline src/templates/rust-all/Makefile.tftpl:27]: cspell and editorconfig?

File: src/templates/rust-all/.github/workflows/generate_docs.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, and run hardhat dodoc on Arrow contracts
```
Comment (ServiceDog) [inline src/templates/rust-all/.github/workflows/generate_docs.yml:1]: This might be better in its own PR

**Review by ServiceDog — APPROVED** [review]

File: src/repositories_rust.tf (hunk 1 above)
Comment (owlot) [reply src/repositories_rust.tf:60]: they are part of the default file list: https://github.com/Arrow-air/tf-github/pull/40/files#diff-1956abf05e8db0150fc12428093f137d1e9cca0a1d2d780a890c054274fb1e30R22 which will be merged into this

File: Makefile (hunk 2 above)
Comment (owlot) [reply Makefile:7]: I don't think they'll be needed for the tf-github repo, although, maybe toml would be helpful to check here too

File: src/templates/all/Makefile (hunk 4 above)
Comment (owlot) [reply src/templates/all/Makefile:7]: This is the 'default' Makefile, it's the minimal that will be deployed to all repo's.
Overrides will be there if more tests are needed (see the Makefile.tftpl specific for rust).

File: src/templates/rust-all/.github/workflows/generate_docs.yml (hunk 6 above)
Comment (owlot) [reply src/templates/rust-all/.github/workflows/generate_docs.yml]: removed from the pr

File: src/templates/rust-all/Makefile.tftpl (hunk 5 above)
Comment (owlot) [reply src/templates/rust-all/Makefile.tftpl]: fixed bf62efa

File: src/repositories_templates.tf (hunk 3 above)
Comment (owlot) [reply src/repositories_templates.tf]: fixed bf62efa

**Review by ServiceDog — DISMISSED** [review]

**Comment by owlot** [comment]
@ServiceDog I added 2 more commits. I noticed I hadn't merged in the local.files map for all repo's created in main.tf and I added a fix for the branch protections which should resolve the issue where you were able to push to personal branches without signing commits.

**Review by ServiceDog — APPROVED** [review]

**Review by benv666 — APPROVED** [review]
